gnosis/eth/account_abstraction/user_operation.py

```diff
@@ -0,0 +1,120 @@
+import dataclasses
+from typing import Any, Dict, Union
+
+from eth_abi import encode as abi_encode
+from eth_typing import ChecksumAddress, HexStr
+from hexbytes import HexBytes
+
+from gnosis.eth.utils import fast_keccak
+
+
+@dataclasses.dataclass
+class UserOperation:
+    """
+    EIP4337 UserOperation for Entrypoint v0.6
+
+    https://github.com/eth-infinitism/account-abstraction/blob/v0.6.0
+    """
+
+    sender: ChecksumAddress
+    nonce: int
+    init_code: bytes
+    call_data: bytes
+    call_gas_limit: int
+    verification_gas_limit: int
+    pre_verification_gas: int
+    max_fee_per_gas: int
+    max_priority_fee_per_gas: int
+    paymaster_and_data: bytes
+    signature: bytes
+    entry_point: ChecksumAddress
+    transaction_hash: bytes
+    block_hash: bytes
+    block_number: int
+    user_operation_hash: bytes
+
+    def __init__(
```

Comment:
I'm thinking that this constructor is a little bit weird because is expected to be used with the `Bundler` response. 
In my opinion could be better a generic `__init__` function with every required parameter (the user operation hash could be optional and calculated in the constructor) and have other `classmethod` to create the instance from the `bundler` response.

Reply:
You are totally right, but as I said, as some parameters will change I'm trying to save us some refactoring time here